### PR TITLE
Novel applications: Private payment

### PR DESCRIPTION
**Pull request by phklive**
In this PR I propose the addition of a `Private payment` file in which we describe:
- The importance of payments in human societies
- The flaws of current payment systems
- The importance of privacy preserving payments
- How can Miden solve the problems of current payment systems
- How does the Miden protocol work at the core to enable those functionalities
- What application can we envision built on top of the Miden protocol leveraging private payments

The final goal is the following:

"Leveraging the Miden primitives mentioned above, we can imagine a powerful payments app that would combine privacy, fast and cheap payments, a stunning user interface, easy to use for a great user experience while being fully permissionless and decentralised."

### CODE REVIEW

File: book/novel_applications/payment/private_payment.html
Comment (Dominik1999) [inline book/novel_applications/payment/private_payment.html]: Let's focus on payments with stables for now. 

We can use this chart https://app.artemisanalytics.com/stablecoins?stablecoinMetric=STABLECOIN_DAILY_TXNS&breakdown=chain&aggregate=MONTH to say that here is a big market. As you say, payments is the proto-use case for blockchains. And it gets traction. 

We can do a short intro about why Ethereum and Bitcoin are not suffice for this (too slow, too expensive). But we can see that Tron, Celo, BnB are valid options. 

But they don't provide privacy. They are only cheap.

File: book/novel_applications/payment/private_payment.html
Comment (Dominik1999) [inline book/novel_applications/payment/private_payment.html]: I would not mention smart contracts here. Let's focus only on payments via stables.

File: book/novel_applications/payment/private_payment.html
Comment (Dominik1999) [inline book/novel_applications/payment/private_payment.html]: Instead of this paragraph I would say, we now see traction for payments because they are so cheap and fast on Tron, Celo, Bnb. 

But those chains don't provide basic privacy nor parallel transaction execution. So they are not made for mass adoption. 

Even centralized systems like Visa or PayPal can only do 100-1000 TPs (https://coinmarketcap.com/academy/glossary/transactions-per-second). 

Let's assume that in 24 hours, 4 billion people will need to make 3 payments (3 payments/day). This translates to 12 billion payments per day, divided by  24 * 60 * 60, resulting in 140000 TPS necessary. 

This level of scalability will only work with client-side proofs.

File: book/novel_applications/payment/private_payment.html
Comment (Dominik1999) [inline book/novel_applications/payment/private_payment.html]: I like this chapter. I would just shorten it a bit and merge this chapter "The importance of privacy in payments" with the first one.

File: book/novel_applications/payment/private_payment.html
Comment (Dominik1999) [inline book/novel_applications/payment/private_payment.html]: For this use case I would not mention programmability.

File: book/novel_applications/payment/private_payment.html
Comment (Dominik1999) [inline book/novel_applications/payment/private_payment.html]: Love this sentence: `Given the limitations of existing blockchains and payments infrastructure, both public and privacy-focused, there is a clear need for a solution that combines privacy, efficiency, and functionality. This is where Miden enters the picture.`

File: book/novel_applications/payment/private_payment.html
Comment (Dominik1999) [inline book/novel_applications/payment/private_payment.html]: We should keep this example step by step guide

Comment (Dominik1999) [reply book/novel_applications/payment/private_payment.html]: Can you maybe link to our transaction blogpost for this as well? 

https://polygon.technology/blog/polygon-miden-transaction-model-2

File: book/novel_applications/payment/private_payment.html
Comment (Dominik1999) [inline book/novel_applications/payment/private_payment.html]: This to me is the third chapter. I like it. 

If you rearrange the previous chapters, I would re-write it here a bit too. 

- Asset transfers are built-in natively into Miden (link to our tutorial), here you can include your step by step guide 
- The Miden protocol scales better
- The Miden protocol provides privacy natively
- Furthermore, the Miden protocol is safer (Account abstraction, assets stored natively in the accounts, recallable transactions)

And again, try to keep it as short as possible.

File: book/novel_applications/payment/private_payment.html
Comment (Dominik1999) [inline book/novel_applications/payment/private_payment.html]: I love it. I think I would only provide one picture to shorten the overall article.

**Review by Dominik1999 — COMMENTED** [review]
This is a first review. I really like what you did here, thanks for the great work. 

In general, I would shorten the article quite a bit. In my mind the structure is as follows (pretty similar to what you have): 

Chapter 1: Payments are the main use case for blockchains, but they are not ready
- Blockchains did not replace traditional payment systems yet
- Bitcoin and Ethereum are too slow and expensive
- Ethereum copy-cats (Tron, Celo, BnB) are cheaper and faster and see significant traction - there is a huge market
- But they have three major problems: don't scale enough (especially not decentralized), don't have privacy, not safe enough for mass adoption (no account abstraction, loss of keys, ...)

Chapter 2: How Miden solves this
- Miden scales better, decentralized and non-custodial (move your chapter here)
- Miden provides privacy natively (as the cheaper option)
- Miden is made for safe payments (account abstraction, assets stored in the account, re-callable transactions)
- [I would not mention programmability and account abstraction here, we can add them later if we think it is not convincing enough]

Chapter 1 and 2 can be fairly short and serve as an intro. 

Chapter 3: Payments on Miden step-by-step
- Let's use your step-by-step guide here `Peer-to-peer private transfers` and link to our tutorial (maybe even the video you or Gobinath made.

Chapter 4: Frontend and Backend
- I like your fronend section. The core message to me here is: The easier and more web2-like the better. Everything should be abstracted away from the user
- Backend: Miden supports payments natively already. The app can pay the fees for its users. Only we need to add a messaging protocol / contact discovery protocol for users

Chapter 5: Your questions can go here.

File: book/novel_applications/payment/private_payment.html
Comment (phklive) [reply book/novel_applications/payment/private_payment.html]: By "more powerful" through smart contracts here I implied that payments had more functionality on Ethereum than on Bitcoin for example using a smart contract you can make payments on certain conditions like an escrow contract or a betting game based on oracles (like Polymarket).

If we want to talk only about account to account direct transfers I can remove this line.

Comment (phklive) [reply book/novel_applications/payment/private_payment.html]: Also stables are smart contracts.

File: book/novel_applications/payment/private_payment.html
Comment (phklive) [reply book/novel_applications/payment/private_payment.html]: I am not entirely sure the given chart presents the message that you have given above. 

SCR-20240913-oemk

Ethereum here seems to suffice considering that it has a volume superior to all the others combined.

Comment (phklive) [reply book/novel_applications/payment/private_payment.html]: Nonetheless, I can rework the introduction: shorten + focus on stables

File: book/novel_applications/payment/private_payment.html
Comment (phklive) [reply book/novel_applications/payment/private_payment.html]: Same question as above: https://github.com/phklive/miden-research/pull/1#discussion_r1758938239

File: book/novel_applications/payment/private_payment.html
Comment (phklive) [reply book/novel_applications/payment/private_payment.html]: I did not fully understand what you meant here: 

> We should keep this example step by step guide

File: book/novel_applications/payment/private_payment.html
Comment (phklive) [reply book/novel_applications/payment/private_payment.html]: When you say 1 picture you mean: 
- 1 picture in the whole article
- 1 picture in this section
- Chose 1 picture between revolut and venmo

Because for now we have many pictures in the article and 3 pictures in this section.

File: book/novel_applications/payment/private_payment.html
Comment (Dominik1999) [reply book/novel_applications/payment/private_payment.html]: I would say, this is not correct. Ethereum has the highest volume with an average transaction of around 100k. But this is not our use case. I would look at the number of transactions. Here Celo seems to be interesting with 16$ average per transaction, also PoS and Tron. 


Screenshot 2024-09-13 at 16 21 46

Those seem to be more human to human payments rather than bigger DeFi transactions as on Ethereum.

File: book/novel_applications/payment/private_payment.html
Comment (Dominik1999) [reply book/novel_applications/payment/private_payment.html]: I would try to keep the use case in the description as simple as possible. So programmability is nice, but let's keep it out for now to keep the use case simple. We can mention it in the last chapter under "further additions".

> Also stables are smart contracts.

On the EVM yes, on Miden not really. Or it depends ...

File: book/novel_applications/payment/private_payment.html
Comment (Dominik1999) [reply book/novel_applications/payment/private_payment.html]: Not sure if the same question applies here. So my hypothesis here is

- if you want a global simple payment system you need 140k TPS (if everyone would use your system)
- no chain currently can provide this
-> chains are not ready for mass adoption

File: book/novel_applications/payment/private_payment.html
Comment (Dominik1999) [reply book/novel_applications/payment/private_payment.html]: I would move this step-by-step guide to chapter 3, in which we explain how a private payment transaction works on Miden. 

Is it clearer now?

File: book/novel_applications/payment/private_payment.html
Comment (Dominik1999) [reply book/novel_applications/payment/private_payment.html]: I would only use one picture in this section. In general, try to use fewer pictures. We can decide on the number of pictures after the next iteration

File: book/novel_applications/payment/private_payment.html
Comment (phklive) [reply book/novel_applications/payment/private_payment.html]: Here you said: 
> Instead of this paragraph I would say, we now see traction for payments because they are so cheap and fast on Tron, Celo, Bnb.
And
> I like this chapter. I would just shorten it a bit and merge this chapter "The importance of privacy in payments" with the first one.

I am not sure what to do

File: book/novel_applications/payment/private_payment.html
Comment (phklive) [reply book/novel_applications/payment/private_payment.html]: Got it.

File: src/novel_applications/payment/private_payment.md
Comment (Dominik1999) [inline src/novel_applications/payment/private_payment.md]: I would name as 1. still scalability. Obviously, Bitcoin and Ethereum don't scale enough to provide fast and cheap payments. But even Tron or Celo with ~100 TPS do not nearly scale enough to provide a payment system for the masses. 

Miden on the other hand, having parallel transaction execution and 1000+ TPS will be able to compete with VISA and PayPal. 

We can list the UI/UX point as third point.

File: src/novel_applications/payment/private_payment.md
Comment (Dominik1999) [inline src/novel_applications/payment/private_payment.md]: When you have changed the points above, I would re-write this sentence to "... that combines scalability, privacy, and functionality"

File: src/novel_applications/payment/private_payment.md
Comment (Dominik1999) [inline src/novel_applications/payment/private_payment.md]: nit: Alice wants to send ~~her~~ 1 Ether to Bob

File: src/novel_applications/payment/private_payment.md
Comment (Dominik1999) [inline src/novel_applications/payment/private_payment.md]: Maybe name "On-chain encrypted notes (on the roadmap)," as the third option and "Signal" and then "Email"? This is a minor point but given the recent discussions around Telegram ...

File: src/novel_applications/payment/private_payment.md
Comment (Dominik1999) [inline src/novel_applications/payment/private_payment.md:78]: I love that list!

File: src/novel_applications/payment/private_payment.md
Comment (Dominik1999) [inline src/novel_applications/payment/private_payment.md]: I think you can skip this account picture and just link to the account page in our docs

https://docs.polygon.technology/miden/miden-base/architecture/accounts/

File: src/novel_applications/payment/private_payment.md
Comment (Dominik1999) [inline src/novel_applications/payment/private_payment.md]: You can leave the mentioning of the Miden VM or winterfell out. We have already abstracted that away from the user. When they use the client, they don't need to know the VM or Winterfell

File: src/novel_applications/payment/private_payment.md
Comment (Dominik1999) [inline src/novel_applications/payment/private_payment.md:122]: nit: keystore (lowercase)

**Review by Dominik1999 — APPROVED** [review]
Looks very good, thanks. I left some comments. Once those are addressed we can merge

File: src/novel_applications/payment/private_payment.md
Comment (phklive) [reply src/novel_applications/payment/private_payment.md:122]: I think that we should keep it Upercased because `Keystore` is the name of the secure element of Android devices:
- Apple -> `Secure enclave`
- Android -> `Keystore`

**Comment by phklive** [comment]
@Dominik1999 please approve my collaborator request + please approve once again to enable merging.

Thanks!

**Review by Dominik1999 — APPROVED** [review]
Already reviewed

**Pull request merged**
